dappy.go

```diff
@@ -67,6 +68,35 @@ func (c client) Auth(username, password string) error {
 	return conn.Bind(results.Entries[0].DN, password)
 }
 
+// Auth implementation for the Client interface
+func (c client) Users() (error, []string) {
+	// establish connection
+	users := []string{}
+	conn, err := connect(c.Host)
+	if err != nil {
+		return err, users
+	}
+	defer conn.Close()
+
+	searchRequest := ldap.NewSearchRequest(
+		c.BaseDN, // The base dn to search
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+		"(&(objectClass=organizationalPerson))", // The filter to apply
+		[]string{"dn", "cn"},                    // A list attributes to retrieve
```

Comment:
if all we want is the user's cn, let's just get this and not dn. 
i think this is definitely a good start, and i wanna merge it after a little touching up (see other comments), BUT, i think we might want to think about adding a configuration field that is a `string[]` of "UserFields" or some such.
Then, we could use that to populate a different kind of `User` struct or perhaps have a user be a `map[string]string` with the fields provided from config as keys and populate the map with the values we get back.
No need to do this now though, just returning cns works. 👍